sql/core/src/test/scala/org/apache/spark/sql/DataFrameRangeSuite.scala

```diff
@@ -127,4 +133,28 @@ class DataFrameRangeSuite extends QueryTest with SharedSQLContext {
       }
     }
   }
+
+  test("Cancelling stage in a query with Range.") {
+    val listener = new SparkListener {
+      override def onTaskStart(taskStart: SparkListenerTaskStart): Unit = {
+        Thread.sleep(100)
```

Comment:
What's the purpose of this listener? Blocking `onTaskStart` only slows down the listener bus thread. It should not block other events from being emitted (It may drop events if the queue is full though). If it does happen, then it's a bug.